docker-entrypoint/dependencies/daemonset/daemonset.go

```diff
@@ -0,0 +1,69 @@
+package daemonset
+
+import (
+	"fmt"
+	"os"
+
+	entry "github.com/stackanetes/docker-entrypoint/dependencies"
+	"github.com/stackanetes/docker-entrypoint/logger"
+	"github.com/stackanetes/docker-entrypoint/util/env"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+type Daemonset struct {
+	name string
+}
+
+func init() {
+	daemonsetsDeps := env.SplitEnvToList(fmt.Sprintf("%sDAEMONSET", entry.DependencyPrefix))
+	if daemonsetsDeps != nil {
+		for dep := range daemonsetsDeps {
+			entry.Register(NewDaemonset(daemonsetsDeps[dep]))
+		}
+	}
+}
+
+func NewDaemonset(name string) (d Daemonset) {
```

Comment:
if you define the name of the return variable in the prototype, then it's already created/allocated:

```
func NewDaemonset(name string) (d Daemonset) {
	d.name = name
	return
}
```

or you can do instead:
```
func NewDaemonset(name string) Daemonset {
	return Daemonset{name: name}
}
```